### PR TITLE
added L0 Tests for JSON variable substitution

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: Tasks/Common/webdeployment-common/variableutility.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+export function isPredefinedVariable(variable: string): boolean {
```
Comment (kmkumaran) [inline Tasks/Common/webdeployment-common/variableutility.ts]: Why did you move this out as separate utility?

Comment (vincentdass) [inline Tasks/Common/webdeployment-common/variableutility.ts]: just moving all the utilities related to variable substitution. It becomes more difficult to mock the entire utility function for L0 tests.

File: Tasks/AzureRmWebAppDeployment/task.json [hunk 2]
```diff
@@ -241,6 +241,7 @@
         "CannotPerformXdtTransformationOnNonWindowsPlatform": "Cannot Perform XDT Transformations on a Non-Windows platform.",
         "XdtTransformationErrorWhileTransforming": "XDT Transformation Error while transforming %s using %s.",
         "PublishusingwebdeployoptionsaresupportedonlywhenusingWindowsagent": "Publish using webdeploy options are supported only when using Windows agent",
-        "WebAppDoesntExist": "Webapp '%s' doesn't exist. Webapp should exist before deployment."
+        "WebAppDoesntExist": "Webapp '%s' doesn't exist. Webapp should exist before deployment.",
+        "EncodeNotSupported": "File Encoding detected to be : %s, which is not supported by Node.js"
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/task.json]: done

File: Tasks/AzureRmWebAppDeployment/Tests/L0JsonVarSub.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    "stats": {
+    	"webAppPkg.zip": {
+    		"isFile": true
+    	}
+    },
+    "osType": {
+        "osType": "Windows"
+    },
+    "checkPath": {
+        "cmd": true
```
Comment (Ajay-MS) [inline Tasks/AzureRmWebAppDeployment/Tests/L0JsonVarSub.ts]: since utiltiy.findFiles method is added you might have to add more answers for checkpath
